src/NuGet.Core/NuGet.ProjectModel/ProjectRestoreMetadata.cs

```diff
@@ -156,17 +156,17 @@ public bool Equals(ProjectRestoreMetadata other)
             }
 
             return ProjectStyle == other.ProjectStyle &&
-                   ProjectPath == other.ProjectPath &&
-                   ProjectJsonPath == other.ProjectJsonPath &&
-                   OutputPath == other.OutputPath &&
-                   ProjectName == other.ProjectName &&
-                   ProjectUniqueName == other.ProjectUniqueName &&
-                   Sources.OrderedEquals(other.Sources.Distinct(), source => source.Source, StringComparer.Ordinal) &&
-                   PackagesPath == other.PackagesPath &&
-                   EqualityUtility.SequenceEqualWithNullCheck(ConfigFilePaths, other.ConfigFilePaths) &&
-                   EqualityUtility.SequenceEqualWithNullCheck(FallbackFolders, other.FallbackFolders) &&
+                   PathUtility.GetStringComparerBasedOnOS().Equals(ProjectPath, other.ProjectPath) &&
+                   PathUtility.GetStringComparerBasedOnOS().Equals(ProjectJsonPath, other.ProjectJsonPath) &&
+                   PathUtility.GetStringComparerBasedOnOS().Equals(OutputPath, other.OutputPath) &&
+                   PathUtility.GetStringComparerBasedOnOS().Equals(ProjectName, other.ProjectName) &&
+                   PathUtility.GetStringComparerBasedOnOS().Equals(ProjectUniqueName, other.ProjectUniqueName) &&
+                   Sources.OrderedEquals(other.Sources.Distinct(), source => source.Source, PathUtility.GetStringComparerBasedOnOS()) &&
```

Comment:
only the UNC sources make sense to be compared based on OS case sensitivity .  http sources shouldn't be treated as case sensitive

Reply:
This is a great catch @rohit21agrawal . 
I'd write a new comparer for the sources to work around that issue.

Reply:
please enlighten me what exactly the issue is here? case sensitive comparison actually makes sense for file based sources  but not for http sources but previous code was making all sources case sensitive not it's case insensitive based on OS.

Besides, if at there will be any issue here, it will be with non-http sources because there won't be different http sources with different casing.

Reply:
agree with @rohit21agrawal. On ubuntu, this will treat `http://unit.test` different from `http://Unit.test`.

Consider making the wrapper around `GetStringComparerBasedOnOS` for sources which uses `GetStringComparerBasedOnOS` for non http sources and a case insensitive one for http sources.